presto-main/src/main/java/com/facebook/presto/tracing/testing/TestingBlock.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tracing.testing;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TestingBlock
+{
+    private final List<TestingPoint> points = new ArrayList<>();
+    private long startTime;
+    private long endTime;
+
+    public TestingBlock(String annotation)
+    {
+        addPoint(new TestingPoint(annotation));
+        this.startTime = System.currentTimeMillis();
+    }
+
+    public long getStartTime()
+    {
+        return startTime;
+    }
+
+    public void end(String annotation)
+    {
+        this.endTime = System.currentTimeMillis();
+        if (startTime > endTime) {
+            throw new RuntimeException("Block start time " + startTime + " is greater than end time " + endTime);
+        }
+        addPoint(new TestingPoint(annotation));
+    }
+
+    public long getEndTime()
+    {
+        return endTime;
+    }
+
+    public synchronized void addPoint(TestingPoint point)
+    {
+        points.add(point);
+    }
+
+    public String toString()
+    {
+        return "{\n" +
+                "  \"startTime\": " + startTime + ",\n" +
+                "  \"endTime\": " + endTime + ",\n" +
+                "  \"points\": " + points.toString() + "\n" +
+                "}";
+    }
```

Comment:
Is this useful? Or we can remove

Reply:
Yeah, will be useful to get the overall tracing structure.